tests-integration/lists_test.go

```diff
@@ -191,3 +191,167 @@ func TestUnreadCountMisordering(t *testing.T) {
 		m.MatchV3DeleteOp(0), m.MatchV3InsertOp(1, roomC), // the unread count decrease coming through
 	)))
 }
+
+func TestBumpEventTypesOnStartup(t *testing.T) {
+	ts := time.Now()
+	t.Log("Alice makes three rooms.")
+	r1CreateState := createRoomState(t, alice, ts)
+	r1Timeline := []json.RawMessage{}
+
+	// Add at least a second between significant events, to ensure there aren't any
+	// timestamp clashes.
+	ts = ts.Add(time.Second)
+	r2CreateState := createRoomState(t, alice, ts)
+	r2Timeline := []json.RawMessage{}
+
+	ts = ts.Add(time.Second)
+	r3CreateState := createRoomState(t, alice, ts)
+	r3Timeline := []json.RawMessage{}
+
+	t.Log("A series of events take place in all three rooms.")
+	// The following sequence of events take place:
+	// (1) r1: topic set
+	// (2) r1: message
+	// (3) r2: message
+	// (4) r3: message
+	// (5) r2: topic
+	// (6) r1: profile change
+
+	ts = ts.Add(time.Second)
+	r1Timeline = append(r1Timeline,
+		testutils.NewStateEvent(
+			t, "m.room.topic", "", alice, map[string]interface{}{"topic": "potato"}, testutils.WithTimestamp(ts),
+		),
+	)
+
+	ts = ts.Add(time.Second)
+	r1Timeline = append(r1Timeline,
+		testutils.NewMessageEvent(t, alice, "message in room 1", testutils.WithTimestamp(ts)),
```

Comment:
I would find this easier to read if this was table-driven and we added a helper to set the timestamp after-the-fact i.e:
```go
ts := time.Now()
steps := []struct{
    timeline *[]json.RawMessage
    event json.RawMessage
}{
    {
        timeline: &r1Timeline,
        event: testutils.NewMessageEvent(t, alice, "message in room 1"),
    },
    // ...
}

for _, step := range steps {
     ts = ts.Add(time.Second)
     step.event = setTimstamp(step.event, ts)
     step.timeline = append(step.timeline, step.event)
}
```

The usage of `*[]json.RawMessage` means you can use `r1Timeline` and have it refer to the new events correctly.